File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        // Algorithm - Simple Binary Search. 
        int start = 0;
        int end = static_cast<int>(nums.size());
        end--;
        
        while (start < end) {
            if (nums[start] < nums[end]) {
                // Already sorted correctly or in a normal array segment
                return nums[start];
            }
            
            int mid = (start + end) / 2;
            
            if (nums[mid] >= nums[end]) {
                start = mid + 1;
            } else {
                end = mid;
            }
        }
        return nums[start];
    }
};

File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        startPtr = 0
        endPtr = len(nums) - 1
        
        while startPtr < endPtr:
            mid = int((startPtr + endPtr)/2)
            
            if nums[mid] > nums[endPtr]:
                startPtr = mid + 1
            else: 
                endPtr = mid
        return nums[startPtr]